File: BigFootVentures/BigFootVentures.Business.Objects/Management/Agreement.cs
namespace BigFootVentures.Business.Objects.Management
{
    public sealed class Agreement : BusinessBase
    {
        #region "Properties"

        public string OwnerName { get; set; }

        public string Name { get; set; }
        public Company BFCompany { get; set; }
        public Company Counterpart { get; set; }
        public string ObjectOfAgreement { get; set; }
        //trademark
        public string OtherRelatedTrademarks { get; set; }
        public string Type { get; set; }
        public string Applicability { get; set; }
        public string DateOfSignature { get; set; }
        public string Comments { get; set; }

        #endregion
    }
}

File: BigFootVentures/BigFootVentures.Application.Web/Scripts/Views/Shared/_Components/Modals/AdvanceSearch/_Agreement.js
function BuildAgreementQuery(query) {
    var name = $('[name="Select.Filter.Agreement.Name"]').val();
    var BFCompany = $('[name="Select.Filter.Agreement.BFCompany"]').val();
    var counterpart = $('[name="Select.Filter.Agreement.Counterpart"]').val();
    var objectOfAgreement = $('[name="Select.Filter.Agreement.ObjectOfAgreement"]').val();
    var type = $('[name="Select.Filter.Agreement.Type"]').val();

    if (name != null && name != undefined && name.length > 0) {
        query += '&name=' + name;
    }

    if (BFCompany != null && BFCompany != undefined && BFCompany.length > 0) {
        query += '&BFCompany=' + BFCompany;
    }

    if (counterpart != null && counterpart != undefined && counterpart.length > 0) {
        query += '&counterpart=' + counterpart;
    }

    if (objectOfAgreement != '') {
        query += '&objectOfAgreement=' + objectOfAgreement;
    }

    if (type != '') {
        query += '&type=' + type;
    }

    return query;
}